nav2_collision_monitor/src/scan.cpp

```diff
@@ -70,10 +70,11 @@ void Scan::getData(
 {
   // Ignore data from the source if it is not being published yet or
   // not being published for a long time
-  if (data_ == nullptr) {
+  auto data_in = std::atomic_load(&data_);
```

Comment:
For all the collision monitor changes: `getData` is triggered by the velocity cmd vel callback. `dataCallback` is triggered by a subscriber that shares the same single threaded executor from the velocity callback. Thus, these are already mutually exclusive and shouldn't require this

Reply:
Thus, I don't think any of the changes to collision monitor are needed?

Reply:
This looked like an issue to me, but only by inspecting the code. I didn't verify it with a test.
I will double-check.

Reply:
At one point awhile ago I went through and made sure we locked everything that needed it knowing we have callback groups setup. I skipped costmap 2D because I assumed that'd been stable since ROS 1, that its probably OK. That was a mistake, but I would be surprised if I missed many others. I'm definitely not perfect, but I did think about some of these before :- )

Reply:
Yes, you are right. Sensors are OK as-is.
I will revert those changes.